python/cugraph/proto/components/scc.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) 2019, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import cudf
+import cugraph
+import numpy as np
+
+
+#
+# TRIM Process:
+#   - removed single vertex componenets
+#   - select vertex with highest out degree
+#   - forwards BFS
+#   - backward BFS
+#   - compute intersection = components
+#   - remove component
+#   - repeat
+
+
+def strong_connected_component(source, destination):
+    """
+    Generate the strongly connected components
+    using the FW-BW-TRIM approach, but skipping the trimming)
+
+    Parameters
+    ----------
+    source : cudf.Seriers
+        A cudf seriers that contains the source side of an edge list
+
+    destination : cudf.Seriers
+        A cudf seriers that contains the destination side of an edge list
+
+    Returns
+    -------
+    cdf : cudf.DataFrame - a dataframe for components
+        df['vertex']   - the vertex ID
+        df['id']       - the component ID
+
+    sdf : cudf.DataFrame - a dataframe with single vertex components
+        df['vertex']   - the vertex ID
+
+    count - int - the number of components found
```

Comment:
This may not converted to the document formatting consistent with the remainder.
```
cdf : cudf.DataFrame
    A data frame for components.

    df['vertex'] : cudf.Series
        The vertex ID.
    df['id'] : cudf.Series
        The component ID
sdf : cudf.DataFrame
    A dataframe with single vertex components.

    df['veretx'] : cudf.Series
        The vertex ID.
count : int
    The number of components found.
```

will lead to consistent formatting. And in the above, I think df['vertex'] and df['id'] (line 47 & 48) should be cdf['vertex'] and cdf['id'] and df['vertex'] in line 51 should be sdf['vertex'].